crates/kumod/src/spf.rs

```diff
@@ -0,0 +1,66 @@
+use config::{from_lua_value, get_or_create_sub_module, serialize_options};
+use kumo_spf::{CheckHostParams, SpfDisposition};
+use mailparsing::AuthenticationResult;
+use mlua::{Lua, LuaSerdeExt, UserData, Value};
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+
+/// A strictly typed subset of [`ConnectionMetaData`] for use in `check_host()`.
+#[derive(Debug, Deserialize)]
+struct SpfConnectionMeta {
+    received_from: SocketAddr,
+}
+
+#[derive(Debug, Deserialize)]
+struct CheckHostInput {
+    domain: String,
+    sender: Option<String>,
+    meta: SpfConnectionMeta,
+}
+
+struct CheckHostOutput {
+    disposition: SpfDisposition,
+    result: AuthenticationResult,
+}
+
+impl UserData for CheckHostOutput {
+    fn add_fields<'lua, F: mlua::UserDataFields<'lua, Self>>(fields: &mut F) {
+        fields.add_field_method_get("disposition", |_, this| Ok(this.disposition.as_str()));
+        fields.add_field_method_get("result", |lua, this| {
+            Ok(lua.to_value_with(&this.result, serialize_options())?)
+        });
+    }
+}
```

Comment:
for this I would be inclined to make the entire output serializable, then `lua.to_value_with` the whole result. It results in less boilerplate here and the resultant native lua table is a bit more flexible to work with in lua if you want to do things using the `pairs` function to iterate it.

```suggestion
#[derive(Serialize)]
struct CheckHostOutput {
    disposition: SpfDisposition,
    result: AuthenticationResult,
}
```